libcontainer/intelrdt/intelrdt.go

```diff
@@ -495,8 +448,14 @@ func IsMBAEnabled() bool {
 
 // Check if Intel RDT/MBA Software Controller is enabled
 func IsMBAScEnabled() bool {
```

Comment:
Looks like this function is unused; looks like in the original implementation it was only used in a test; https://github.com/opencontainers/runc/commit/95af9eff82157317817d3d550626f3e5db240d8a#diff-10a177e788e7be5a6826c74946ebbb7390ead3a61f1342a4776dba4d57143942R86

```go
func TestIntelRdtSetMemBwScSchema(t *testing.T) {
 	if !IsMbaScEnabled() {
 		return
 	}
```

But not outside of that (only the related `mbaScEnabled` variable was used)


That check was removed in https://github.com/opencontainers/runc/commit/1b4c30fd8b719834a5492a17893994a996c0a7ce (https://github.com/opencontainers/runc/pull/2920), so it looks unused indeed

Also checked uses outside of runc itself, but it looks like there's none; https://grep.app/search?q=IsMbaScEnabled

Reply:
Looking further at this; if we would remove this function, then `findResctrlMountinfo()` is only used in a single place, as part of `func Root()`; which _also_ wraps it in a `sync.Once`;

```
// Root returns the Intel RDT "resource control" filesystem mount point.
 func Root() (string, error) {
 	rootOnce.Do(func() {
```

So in that case, we can remove the `findMountinfoOnce.Do` and associated variables for `findResctrlMountinfo()`

Reply:
The other reviewers also noted that `IsMBAScEnabled` is not used in searchable public source code. I am not convinced it is proof enough that the function can be deleted without breaking anybody's code. Intel RDT is very much an enterprise datacenter feature so it's entirely possible that some company's private code is consuming it. Could we defer the removal of this function to a follow-up PR?

Reply:
I think the relevant point here is that `libcontainer/` is considered not to be for external consumption; there's some remaining uses of it outside of runc itself (most of that for historical reasons), but the intent is to better defined "externally consumable" packages (if any) and "internal" packages; see https://github.com/opencontainers/runc/issues/3028

Reply:
Ahh, I couldn't remember if it was all of libcontainer or just certain subpackages

Reply:
Yes, I think the docs / readme should probably more clear on that once we settle "what bits" (some bits are a bit of a "grey" area though)